src/Facets/EcoFacet.sol

```diff
@@ -139,14 +139,23 @@ contract EcoFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable, LiFiData {
     {
         _validateEcoData(_bridgeData, _ecoData);
 
-        _bridgeData.minAmount = _depositAndSwap(
+        uint256 actualAmountAfterSwap = _depositAndSwap(
             _bridgeData.transactionId,
             _bridgeData.minAmount,
             _swapData,
-            payable(msg.sender),
-            0
+            payable(msg.sender)
         );
 
+        if (actualAmountAfterSwap > _bridgeData.minAmount) {
+            uint256 positiveSlippage = actualAmountAfterSwap -
+                _bridgeData.minAmount;
+            LibAsset.transferAsset(
```

Comment:
and I think we can use `transferERC20` instead of `transferAsset` as we already know it's not native asset having `noNativeAsset` modifier above